pydantic/schema.py

```diff
@@ -135,7 +146,10 @@ def model_schema(
     if model_name in nested_models:
         # model_name is in Nested models, it has circular references
         m_definitions[model_name] = m_schema
-        m_schema = {'$ref': ref_prefix + model_name}
+        if ref_template:
+            m_schema = {'$ref': ref_template.substitute(model_name=model_name)}
+        else:
+            m_schema = {'$ref': ref_prefix + model_name}
```

Comment:
This could probably be done more elegantly.

Some additional things I could do.
1. regular `ref_prefix`s could also be resolved using the same `string.Template` method. `"#/components/schemas/${model_name}"`
2. better error message if a bad template is passed.
3. coerce a regular string into a template instead of taking an actual `string.Template` objecty.